docs/explanation/parts-lifecycle.rst

```diff
@@ -3,4 +3,261 @@
+    * - ``parts/<part-name>/src``
+      - ``CRAFT_PART_SRC``
+      - The location of the source after the pull step.
+    * - ``parts/<part-name>/build``
+      - ``CRAFT_PART_BUILD``
+      - The working directory during the build step.
+    * - ``parts/<part-name>/install``
+      -  ``CRAFT_PART_INSTALL``
+      - Contains the results of the build step and the stage packages. It's also the
+        directory where the ``organize`` event renames the built files.
+    * - ``stage``
+      - ``CRAFT_STAGE``
+      - Shared by all parts, this directory contains the contents of each part's
+        ``CRAFT_PART_INSTALL`` after the stage step. It can contain development
+        libraries, headers, and other components (such as pkgconfig files) that need to
+        be accessible from other parts.
+    * - ``CRAFT_PRIME``
+      - ``prime``
+      - Shared by all parts, this directory holds the final components after the prime
+        step.
+    * - The current project's path in the filesystem.
+      - ``CRAFT_PROJECT_DIR``
+      - Used to access resources from the project's subtree, such as an icon or version
+        file.
+
+
+Overriding a step
+-----------------
+
+Each plugin defines the default actions that happen during a step. This behavior can be
+changed in two ways:
+
+- With the ``override-<step-name>`` key. See `Override build steps
+  <https://snapcraft.io/docs/scriptlets>`_ for guidance.
+- (core20 only) With a local plugin, which can inherit the parent plugin or scaffolding
+  from the original. See `Writing local plugins
+  <https://snapcraft.io/docs/writing-local-plugins>`_ for details.
+
+
+.. _explanation-parts-lifecycle-processing-order:
+
+Processing order and dependencies
+---------------------------------
+
+Each lifecycle step depends on the completion of the previous step. During build,
+Snapcraft walks through one step at a time, processing that step for all parts. Within a
+step, the parts are processed in alphabetical order. Only after a step is complete for
+every part will it continue to the next step.
+
+
+Overriding the part order
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The part order can be overriden by the ``after`` key in the part's definition. The
+purpose of the key is to stagger the part order so that interrelated parts can provide
+data to each other.
+
+With ``after``, the part order follows modified rules:
+
+- Parts are ordered alphabetically by name, as usual.
+- When the build reaches a part that another depends on, the dependent part will only
+  start its build and stage steps after the initial part finishes its stage step.
+- After a chain of parts completes, the step continues to the next part in alphabetical
+  order.
+
+
+Example 1 -- Default lifecycle
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this example, the default order applies. Each step is processed serially, and within
+each step the parts are processed in alphabetical order.
+
+.. code-block:: yaml
+    :caption: snapcraft.yaml
+
+    parts:
```

Comment:
Is 4 spaces a standard or is it formatted automatically? Asking because with such short part names, there's no overlap and it looks strange.

![image](https://github.com/user-attachments/assets/97688cb8-2cae-49d7-9a9b-f05e034f950c)

Reply:
Good catch. Here's what it looks like with corrected indentation and less terse nonce names:

![image](https://github.com/user-attachments/assets/53c239ce-58f6-4e92-9bd1-1f59c6763c5e)

Do you think the order is harder to parse with the names?

Reply:
Looks good - it’s also easier to notice the order with names like that.